persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java

```diff
@@ -325,6 +329,9 @@ private long timeHeaderToSeconds(String header) {
 
    @Override
    public void process(KeyFilter keyFilter, final CacheLoaderTask cacheLoaderTask, Executor executor, boolean loadValue, boolean loadMetadata) {
+      if (!started) {
```

Comment:
I don't think this is sufficient to protect the initial key retrieval as it will be propagated to the caller.

Reply:
The test that was added only catches things after it starts producing entries but not before.

Reply:
I kinda wonder if we just want to add a semaphore to this, SFS and SIFS instead (like rocks and level db). We could even have a wrapper class that guarantees orderly shutdown (and apply this to user added custom stores as well).